flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/Observer.java

```diff
@@ -94,6 +96,14 @@ private void observeJmDeployment(
                     flinkApp.getMetadata().getNamespace(),
                     status);
 
+            List<DeploymentCondition> conditions = status.getConditions();
+            for (DeploymentCondition dc : conditions) {
+                if ("FailedCreate".equals(dc.getReason())
+                        && "ReplicaFailure".equals(dc.getType())) {
+                    throw new DeploymentFailedException(
+                            DeploymentFailedException.COMPONENT_JOBMANAGER, dc);
+                }
+            }
```

Comment:
We should not check for new Errors if we are already in ERROR deploymant status, otherwise it wont be possible to submit a new spec to fix the job.

Reply:
Done, exception is now only raised when not already in ERROR